boards/riscv/tlsr9518adk80d/pinmux.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2021 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gpio.h>
+#include <drivers/pinmux.h>
+
+
+/**
+ *  @brief  Define UART0 pins : TX(PA3 PB2 PD2), RX(PA4 PB3 PD3)
+ */
+#define UART0_TX_PA3            GPIO_PA3
+#define UART0_TX_PA3_FUNC       PINMUX_FUNC_B
+
+#define UART0_TX_PB2            GPIO_PB2
+#define UART0_TX_PB2_FUNC       PINMUX_FUNC_C
+
+#define UART0_TX_PD2            GPIO_PD2
+#define UART0_TX_PD2_FUNC       PINMUX_FUNC_A
+
+
+#define UART0_RX_PA4            GPIO_PA4
+#define UART0_RX_PA4_FUNC       PINMUX_FUNC_B
+
+#define UART0_RX_PB3            GPIO_PB3
+#define UART0_RX_PB3_FUNC       PINMUX_FUNC_C
+
+#define UART0_RX_PD3            GPIO_PD3
+#define UART0_RX_PD3_FUNC       PINMUX_FUNC_A
+
+
+/**
+ *  @brief  Define UART1 pins: TX(PC6 PD6 PE0), RX(PC7 PD7 PE2)
+ */
+#define UART1_TX_PC6            GPIO_PC6
+#define UART1_TX_PC6_FUNC       PINMUX_FUNC_C
+
+#define UART1_TX_PD6            GPIO_PD6
+#define UART1_TX_PD6_FUNC       PINMUX_FUNC_A
+
+#define UART1_TX_PE0            GPIO_PE0
+#define UART1_TX_PE0_FUNC       PINMUX_FUNC_B
+
+
+#define UART1_RX_PC7            GPIO_PC7
+#define UART1_RX_PC7_FUNC       PINMUX_FUNC_C
+
+#define UART1_RX_PD7            GPIO_PD7
+#define UART1_RX_PD7_FUNC       PINMUX_FUNC_A
+
+#define UART1_RX_PE2            GPIO_PE2
+#define UART1_RX_PE2_FUNC       PINMUX_FUNC_B
+
+
+/**
+ *  @brief  Config board pins
+ */
+static int b91_pinmux_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	const struct device *pinmux =
+		device_get_binding(DT_LABEL(DT_NODELABEL(pinmux)));
```

Comment:
No need for device_get_binding or label properties anymore.

```c
const struct device *pinmux = DEVICE_DT_GET(DT_NODELABEL(pinmux));
```

If it is possible that the pinmux device may fail to initialize, this must be followed with:

```c
if (!device_is_ready(pinmux)) {
        return -ENODEV;
}
```

Reply:
Fixed. Please pay attention that pins configuration is moved to DTS (see Serial driver).